packages/mui-joy/src/Button/Button.tsx

```diff
@@ -219,7 +219,7 @@ const Button = React.forwardRef(function Button(inProps, ref) {
   const size = inProps.size || buttonGroup.size || sizeProp;
   const color = inProps.color || buttonGroup.color || colorProp;
   const disabled =
-    (inProps.disabled || inProps.loading) ?? (buttonGroup.disabled || disabledProp || loading);
+    (inProps.loading || inProps.disabled) ?? (buttonGroup.disabled || disabledProp || loading);
```

Comment:
I believe it should be the same in the other condition:
```suggestion
    (inProps.loading || inProps.disabled) ?? (buttonGroup.disabled || loading || disabledProp);
```

Reply:
It doesn't change anything here. Order in the left part is important because we rely on the difference between `false` and `undefined` (hence `??` and not `||`).

`undefined || undefined === undefined`
`undefined || false === false`
`false || undefined === undefined`

`(inProps.disabled || inProps.loading)` didn't work correctly for the case when `disabled` was `false`, and `loading` was `undefined`: it fell to the right part and used the one of the ButtonGroup. With `(inProps.loading || inProps.disabled)` it works just fine, because if `loading` is `false` and `disabled` is `undefined`, we should use the right part.

Reply:
Yes, I had understood your change. The change I mentioned isn't about your fix. It's regarding the behavior when using [theme default props](https://mui.com/joy-ui/customization/themed-components/#theme-default-props). When the theme's `loading` property is true, the button should always be disabled. If the theme's `loading` property is `false` or `undefined`, then it should check the theme's default props for the `disabled` property if defined, similar to what you described for the component props in the PR description, not the theme's default props. This isn't related to your issue, but we can improve the logic in this PR.

Reply:
I'm still sure that we won't improve anything by doing that: changing the order inside `buttonGroup.disabled || disabledProp || loading` won't make any difference. If any of them is `true`, the result will be `true` regardless of what comes first.
If all of them are `false` or `undefined`, the result will be negative, and by changing this order we will only switch the cases when the negative result is `false` and when it's `undefined`.

The order of the left part does change the logic, while the order of the right part doesn't. Are you talking about readability only? I mean, I don't mind changing the order, but I want to understand what I'm missing.

Reply:
It will short-circuit and skip evaluating the right value. So, once `loading` is true, `disabledProp` won't be evaluated, which is more efficient. In the case where the button is loading and should be disabled, it will stop at the first `true` value, making it more performant.

For example:
- `false || true || false` → more efficient
- `false || false || true` → same, but less efficient.

And yes, as you mentioned, it also improves readability.

Reply:
Well, in that case we will lose in other cases, won't we? I mean if `loading` is `false` or `undefined`, but `disabledProp` is `true`.
Anyway, I believe it's not significant in terms of performance, but if you feel that readability is improved, it's good enough. Fixed with the latest commit.

Reply:
> Well, in that case we will lose in other cases, won't we?

Oh yeah, you are right. I didn't account for that.

Yes, so the sole purpose for the change is readability. Left part matching the right part.